pkg/datastore/postgres/repositories/header_repository_test.go

```diff
@@ -442,4 +443,26 @@ var _ = Describe("Block header repository", func() {
 			Expect(missingBlockNumbers).To(ConsistOf([]int64{2, 4}))
 		})
 	})
+
+	Describe("GetMostRecentHeaderBlockNumber", func() {
+		It("gets the most recent header block number", func() {
+			_, createHeader1Err := repo.CreateOrUpdateHeader(header)
+			Expect(createHeader1Err).NotTo(HaveOccurred())
+
+			header2BlockNumber := header.BlockNumber + int64(1)
+			header2 := fakes.GetFakeHeader(header2BlockNumber)
+			_, createHeader2Err := repo.CreateOrUpdateHeader(header2)
+			Expect(createHeader2Err).NotTo(HaveOccurred())
+
+			mostRecentHeaderBlock, err := repo.GetMostRecentHeaderBlockNumber()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(mostRecentHeaderBlock).To(Equal(header2BlockNumber))
+		})
+
+		It("returns an error if it fails to get the most recent header", func() {
+			_, err := repo.GetMostRecentHeaderBlockNumber()
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError(sql.ErrNoRows))
```

Comment:
This seems like potentially an issue for freshly deployed systems as well, if execute starts before header sync. Wondering if we want to handle `sql.ErrNoRows` in the storage watched from both of these repository calls - seems like maybe we could delay + retry, set minID to 0, or something